Assets/Labrys/FeatureLibrary.cs

```diff
@@ -0,0 +1,195 @@
+using Labrys.FeatureEditor;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+#if UNITY_EDITOR
+using UnityEditor;
+#endif
+using UnityEngine;
+
+namespace Labrys
+{
+    public class FeatureLibrary : ScriptableObject, IEnumerable<FeatureLibrary.Entry>, ISerializationCallbackReceiver, ICloneable
+    {
+        [field: SerializeField]
+        public string TargetDirectory { get; private set; }
+
+        private Dictionary<string, ushort> nameToFAID;
+        private FeatureAsset[] features;
```

Comment:
(for my own understanding) In short:
Adding a FeatureAsset to the Library essentially adds it to the end of an array, with an additional search structure which indexes by name (that is specified on add). Accessing a FA by index is trivial, accessing by name has a lookup in the name->index Dictionary.

Reply:
Additional note now that I'm looking at this more: `nameToFAID.Values()` will only contain sequential integers. I think you can just replace this with an array of strings of the same size as `features`, since their order will be equivalent (purely for storage). The Dictionary is necessary because of the need for a reverse lookup from names->assets.